Client/mods/deathmatch/logic/rpc/CVehicleRPCs.cpp

```diff
@@ -653,3 +654,17 @@ void CVehicleRPCs::SetVehiclePlateText(CClientEntity* pSourceEntity, NetBitStrea
         }
     }
 }
+
+void CVehicleRPCs::SpawnVehicleFlyingComponent(CClientEntity* const sourceEntity, NetBitStreamInterface& bitStream)
+{
+    CClientVehicle* vehicle = m_pVehicleManager->Get(sourceEntity->GetID());
+    if (!vehicle)
+        return;
+
+    std::uint8_t nodeIndex;
+    std::uint8_t collisionType;
+    std::int32_t removalTime;
+
+    if (bitStream.Read(nodeIndex) && bitStream.Read(collisionType) && bitStream.Read(removalTime))
+        vehicle->SpawnFlyingComponent(static_cast<eCarNodes>(nodeIndex), static_cast<eCarComponentCollisionTypes>(collisionType), removalTime);
```

Comment:
maybe early continue?

Reply:
There is no need, this is just one short condition. The guidelines are to avoid nesting conditions within each other, not to avoid them altogether

Reply:
It would make it much more readable though:
```cpp
if (!bitStream.Read(nodeIndex))
    return;
if(!bitStream.Read(collisionType))
    return;
if(!bitStream.Read(removalTime))
    return;
vehicle->SpawnFlyingComponent(static_cast<eCarNodes>(nodeIndex), static_cast<eCarComponentCollisionTypes>(collisionType), removalTime);

```

Reply:
An example of inappropriate use of early return. This make noise instead of gaining readability.

Reply:
Disagree. This makes the code much more cleaner and easier to maintain.
Having very long lines, long if statements creates chaos.